examples/HelloWorldExampleDS/HelloWorldSubscriber.cpp

```diff
@@ -18,105 +18,118 @@
         descriptor->add_listener_port(rdn(gen)); // IANA ephemeral port number
-
         descriptor->wait_for_tcp_negotiation = false;
-        PParam.rtps.userTransports.push_back(descriptor);
+        participant_qos.transport().user_transports.push_back(descriptor);
     }
     else
     {
-        if(!IsAddressDefined(server_address))
+        if (!IsAddressDefined(server_address))
         {
             server_address.kind = LOCATOR_KIND_UDPv4;
             server_address.port = default_port;
             IPLocator::setIPv4(server_address, 127, 0, 0, 1);
         }
 
         ratt.metatrafficUnicastLocatorList.push_back(server_address);
-        PParam.rtps.builtin.discovery_config.m_DiscoveryServers.push_back(ratt);
+        participant_qos.wire_protocol().builtin.discovery_config.m_DiscoveryServers.push_back(ratt);
     }
 
-    mp_participant = Domain::createParticipant(PParam);
+    participant_qos.wire_protocol().builtin.discovery_config.discoveryProtocol = DiscoveryProtocol_t::CLIENT;
+    participant_qos.wire_protocol().participant_id = 3;
+    participant_qos.wire_protocol().builtin.discovery_config.leaseDuration = eprosima::fastrtps::c_TimeInfinite;
+    participant_qos.name("Participant_sub");
+
+    mp_participant = DomainParticipantFactory::get_instance()->create_participant(0, participant_qos);
 
     if (mp_participant == nullptr)
     {
         return false;
     }
 
-    //REGISTER THE TYPE
-
-    Domain::registerType(mp_participant,&m_type);
-    //CREATE THE SUBSCRIBER
-    SubscriberAttributes Rparam;
-    Rparam.topic.topicKind = NO_KEY;
-    Rparam.topic.topicDataType = "HelloWorld";
-    Rparam.topic.topicName = "HelloWorldTopic";
-    Rparam.topic.historyQos.kind = KEEP_LAST_HISTORY_QOS;
-    Rparam.topic.historyQos.depth = 30;
-    Rparam.topic.resourceLimitsQos.max_samples = 50;
-    Rparam.topic.resourceLimitsQos.allocated_samples = 20;
-    Rparam.qos.m_reliability.kind = RELIABLE_RELIABILITY_QOS;
-    Rparam.qos.m_durability.kind = TRANSIENT_LOCAL_DURABILITY_QOS;
-    mp_subscriber = Domain::createSubscriber(mp_participant,Rparam,(SubscriberListener*)&m_listener);
-
-    if (mp_subscriber == nullptr)
+    TypeSupport ts(new HelloWorldPubSubType());
+
+    mp_participant->register_type(ts);
+
+    TopicQos topic_qos = TOPIC_QOS_DEFAULT;
+
+    topic_qos.history().depth = 30;
+    topic_qos.history().kind = KEEP_LAST_HISTORY_QOS;
+    topic_qos.resource_limits().max_samples = 50;
+    topic_qos.resource_limits().allocated_samples = 20;
+
+    SubscriberQos subscriber_qos = SUBSCRIBER_QOS_DEFAULT;
+    DataReaderQos datareader_qos = DATAREADER_QOS_DEFAULT;
+
+    datareader_qos.reliability().kind = RELIABLE_RELIABILITY_QOS;
+    datareader_qos.durability().kind = TRANSIENT_LOCAL_DURABILITY_QOS;
+
+    mp_subscriber = mp_participant->create_subscriber(subscriber_qos);
+
+    Topic* topic = mp_participant->create_topic("HelloWorldTopic", "HelloWorld", topic_qos);
```

Comment:
I suggest instead of hardcoding the type name, obtaining it from the TypeSupport API